### ISSUE TITLE
Photo specific sidebar in Gallery

### ISSUE THREAD

**Issue by karlitschek** [nextcloud/gallery#514]
The Gallery app should have a left sidebar to find and view photos similar to other photo apps on Android, iOS, Mac, Windows, Gnome, KDE, ...

Sidebar entries could be:
- All photos (in all folders as timeline)
- Albums (grouped by folders as today)
- Recently imported
- Places (worldmap with pins of photos based on exif data)
- People (Different people with face detection (once we have the technology))
- Memories (automatic grouping of photos from similar location and at a similar time)
- more based on inspiration from other photo apps.

**Comment by karlitschek** [comment]
@jancborchardt @MorrisJobke @oparoz What do you think?

**Comment by jancborchardt** [comment]
Yep, absolutely think that’s the way to go for Gallery. :+1: Currently it’s just a weird other folder-based view, but if we make it a timeline, sort everything by date, and add these further enhancements which other photo apps also have, this is way more proper. (And then it won’t be just a "slightly different grid view".)

I guess the main point is that currently the Gallery is quite outdated and doesn’t follow a lot of the design standards we developed, and has the folder-based view. So it would be (don’t shoot me :D) a good point in time to actually do it properly with Vue.js and the design guidelines. What do you think @skjnldsv @juliushaertl?

There’s also some more work being done on [the Maps app](https://github.com/nextcloud/maps/pull/44) by @eneiluj including showing pictures on the map – so we could use that vice versa in the Gallery app too to show a map for the pictures.

**Comment by juliushaertl** [comment]
I absolutely agree with Jan that this is a good point in time to rework the gallery app. There is a lot of code in it that is basically copied from the files_sharing app and therefore utterly outdated, because we changed a lot there recently for a better user experience. 

For the sidebar and proper filtering, we somehow depend on https://github.com/nextcloud/server/issues/5106

> All photos (in all folders as timeline)

Not sure if we can achieve that already? Is there a dav search that we can perform to get all images in all folders?

> Recently imported

Same as for all photos.

> Places (worldmap with pins of photos based on exif data)

Not sure how the maps app looks for photos and caches the exif data, but we should probably just integrate the view here then.

> Memories (automatic grouping of photos from similar location and at a similar time)
more based on inspiration from other photo apps.

A candidate once we have https://github.com/nextcloud/server/issues/5106

**Comment by skjnldsv** [comment]
Once we have the new sidebar standard, we might clone the thing into a OCA.Gallery.Sidebar with the same behaviour?

It should simplify our process a lot and allow further updates and security fixes in a single location :)

**Comment by jakobroehrl** [comment]
Some more ideas/wishes ;)

 - It would be great if the user can select photos for performing a action (delete, archive, share, ...) to multiple photos at once. 
 - An archive (these photos are hidden from the timeline)
 - Collage creation 
 - videos should although shown

What about the mobile side? 
The current situation is, that the photos need a loooot of time to load on android (although previewgenerator)

**Comment by jancborchardt** [comment]
@jakobroehrl those are out of scope for this specific issue – please open separate issues for separate proposals (also searching existing issues if there’s already related ones). Thanks! :)

**Comment by jancborchardt** [comment]
So should we put that high on the list for Nextcloud 18 directly? We can do some planning in this cycle and see at the Contributor Week what we need to set it up.

**Comment by karlitschek** [comment]
sure. let's plan this

**Comment by janis91** [comment]
I like that idea. A few weeks ago I started with some small frontend prototype (with react, because I'm not that familiar with vue yet). That perhaps gives some ideas. The prototype already includes things like a lightbox, lazy loading, responsive features, etc. But it is definitely a prototype so far. I did not have the time to connect that to the existing gallery API or to integrate Open Street Map yet.
<img width="1674" alt="sc1" src="https://user-images.githubusercontent.com/20393872/62295392-3aa77a00-b46d-11e9-9001-6bfa116bd26a.png">
<img width="1660" alt="sc2" src="https://user-images.githubusercontent.com/20393872/62295393-3aa77a00-b46d-11e9-9c31-0a2d7407c310.png">

The repo is private, because I don't want to publish this project (I think I won't have the time for it, to fulfill my own standards), but I can give you access to it, if you like.

**Comment by jancborchardt** [comment]
@janis91 very cool, looks nice! :)
In Vue we do have the benefit of a lot of components already being available: https://github.com/nextcloud/nextcloud-vue – for example a Viewer which is basically the lightbox, or the left navigation which uses the Nextcloud styleguide.
cc @nextcloud/designers about that.

Also seeing you are in Stuttgart, maybe come over at the next Nextcloud Contributor Week (we just had one a month ago, but doing it every ~4 months)? :)

**Comment by janis91** [comment]
>Also seeing you are in Stuttgart, maybe come over at the next Nextcloud Contributor Week (we just had one a month ago, but doing it every ~4 months)? :)

I will look out for it. 

>In Vue we do have the benefit of a lot of components [...]

I understand the benefit, it's just that I did not have the opportunity to learn Vue yet ;-)

**Comment by Cyber1000** [comment]
Besides from the renewal of the gallery app, which is a good thing by the way, would it make sense for you to create own albums (you may share with others) within the gallery app? (I've asked in https://help.nextcloud.com/t/is-there-a-good-app-to-organize-photos/57742 and was told to read this issue). It can't be only me who would like this feature, or are there already solutions I didn't find so far?

**Comment by karlitschek** [comment]
@janis91 This is super cool! It would be great if you could help to push this forward. vue is prefered as @jancborchardt mentioned :-)

**Comment by janis91** [comment]
> @janis91 This is super cool! It would be great if you could help to push this forward. vue is prefered as @jancborchardt mentioned :-)

My feeling is, that this will require a lot changes, @jancborchardt does it make sense to create a new branch and start from scratch in it, or what do you think?

**Comment by Spartachetto** [comment]
Now we have a functioning maps app, thanks to @eneiluj and the others who worked on it.

It would be really interesting to have an interaction between gallery and maps: if I have both apps active I'd love to have the possibility to switch from one view to the other...!

I think that the request is **NOT** simple but it would be a terrific improvement...

**Comment by jancborchardt** [comment]
@janis91 @skjnldsv since we are doing everything in Vue now, it would absolutely make sense to do the Gallery in Vue too if we are making changes anyway.

Also design-wise, things to look at are:
- [Instagram](https://www.instagram.com/)
- [Google Photos](https://www.google.com/photos/about/)
- [macOS Photos](https://www.apple.com/macos/photos/)
- [Arcane Photos](https://photos.arcaneoffice.com/)
